core/domain/user_jobs.py

```diff
@@ -233,6 +269,40 @@ def map(item):
                                 'type': 'exploration',
                                 'id': item.id
                             })
+        elif isinstance(item, collection_models.CollectionRightsModel):
+            if item.deleted:
```

Comment:
Feels like a lot of this duplicates the previous clause and can be refactored into a separate private method; wdyt?